packages/nimble-components/src/table/specs/table-column-specs/table-column-menu-button.md

```diff
@@ -0,0 +1,203 @@
+        -   Handled by table rows. When handling this event, a table row will fire the `row-slots-request` event to request the table to slot the necessary elements within the row.
+        -   Event details:
+            <!-- prettier-ignore -->
+            ```ts
+            interface CellViewSlotRequestEventDetail {
+                slots: { slot: string, name: string }[];
+            }
+            ```
+    -   `row-slots-request` event
+        -   Fired by table rows to request that column's slots to be placed within that row. This event is fired in response to the row recieving a `cell-view-slots-request` event fired by one of its cell views.
+        -   Handled by table. When handling this event, the table will update its template to move the appropriate slots into the correct table row.
+        -   Event details:
+            <!-- prettier-ignore -->
+            ```ts
+            interface RowSlotRequestEventDetail {
+                columnInternalId: string;
+                rowId: string;
+                slots: { slot: string, name: string }[];
+            }
+            ```
+
+## Alternative Implementations / Designs
+
+_Alternative #1_
+
+Rather than creating a menu button column, we could create a select column. However this has a number of challenges steming from the fact that the `nimble-select` has a different slotting model than the `nimble-menu-button`. As a result, it would be very difficult have an API that allows providing unique options for each record without a significant redesign of the select. Additionally, the `nimble-select` has interactions that allow the value to be changed without the component being open. This poses a challenge of handling all the cases where a given select's options need to be updated.
+
+## Open Issues
+
+-   In the Software Table, how should the currently installed version be represented? Currently it is italicized in both the menu and menu-button, but this is probably not the solution we want in the nimble table column.
+-   In the Software Table, the full software version is displayed as the title of the menu button and menu items. Where should that information go?
+-   There is no visual design spec for this feature yet. Therefore there are a few open visual design questions, including the following:
+    -   If we only support one appearance for the menu-button, is `ghost` the appropriate one to support?
+    -   Is there any concern that the button text will not align with the column header because of the padding between the edge of the button and the text within the button?
+-   If the menu is open in a cell, it needs to be closed when scrolling. However, the `focusedRecycleCallback()` doesn't provide the appropriate hook to do this today. The problem appears to be that the menu has focus when the menu button is open, which means the focus isn't actually in the cell view -- it's in the menu slotted into the cell view. We need to come up with a plan to address this.
```

Comment:
I have a [branch](https://github.com/ni/nimble/commit/371654af0e660176adb8d1ead5ed447cebef5707) / [Storybook](https://60e89457a987cf003efc0a5b-hbjvjoxggm.chromatic.com/?path=/story/components-table-column-menu-button--menu-button-column) that resolves the arrow key conflicts with the menu button column. So if we like the behavior this way, we don't necessarily need to switch to focusing the cell instead of the menu button.  
(See commit comment for what I changed)

Reply:
A few thoughts:
- It seems like there is a bug that will need to be resolved if we go that direction. The first time a menu button is focused in the table up/down arrows open the menu rather than move focus to a different cell. After you've been navigating in the table, it seems to work fine.
- I have some concerns about having up/down arrows not open the menu when the menu button is focused. That would go against ARIA guidelines.

Reply:
Ah, good to know that the DownArrow/UpArrow to open the menu is ARIA guidance.

In addition to the bug you mentioned (seems to just happen if the menu button is focused via Tab), I also noticed that there's scrolling issues with that build I linked - once a menu button is focused, mousewheel won't let you scroll past the menu button (it stops scrolling once it's the first/last visible row).

Probably letting the cell be focused (and getting rid of the 'single interactive element' code path in the keyboard nav code) is still the least amount of work to get everything behaving in a consistent way.